[KTIJ-4113] - Enabling align when multiline option for chained methods

DanielFidalgo opened the pull request:
This PR makes the option 'Align when multiline' available for chained method calls that are wrapped.
Continuation indent still takes precedence and the option being anabled will only work when continuation indent is disabled.

dimonchik0036 reviewed (changes requested):
Please add tests for many different cases. Such changes should be well covered.
Also, the commit message is not properly formatted. Should be something like this:
```
[kotlin][formatter] Enable align when multiline option for chained methods

^KTIJ-4113 Fixed
```
[CONTRIBUTING.md](https://github.com/JetBrains/intellij-community/blob/master/plugins/kotlin/CONTRIBUTING.md#3-pull-request-requirements)

DanielFidalgo commented:
> Please add tests for many different cases. Such changes should be well covered. Also, the commit message is not properly formatted. Should be something like this:
> 
> ```
> [kotlin][formatter] Enable align when multiline option for chained methods
> 
> ^KTIJ-4113 Fixed
> ```
> 
> [CONTRIBUTING.md](https://github.com/JetBrains/intellij-community/blob/master/plugins/kotlin/CONTRIBUTING.md#3-pull-request-requirements)

Added test file and fixed commit message

dimonchik0036 reviewed (changes requested):
In Kotlin, we also have safe call chains (`a?.b`). Could you please add tests for this case too?
They can be mixed with regular dot calls in any order

DanielFidalgo commented:
> In Kotlin, we also have safe call chains (`a?.b`). Could you please add tests for this case too? They can be mixed with regular dot calls in any order

Added test file for safe access chain call testing

DanielFidalgo commented:
And I closed the PR thinking I was closing the change request 🤦

DanielFidalgo commented:
Opened new pull request on https://github.com/JetBrains/intellij-community/pull/2540 in case this PR cannot be reopened
